[Bug]: Gemma3 Batch Offline Inference Error: Cascade attention does not support sliding window

### Your current environment

<details>
<summary>Gemma 3 can't use sliding window attention when doing offline batch inference</summary>

```text
Collecting environment information...
PyTorch version: 2.6.0+cu124
Is debug build: False
CUDA used to build PyTorch: 12.4
ROCM used to build PyTorch: N/A

OS: Ubuntu 22.04 LTS (x86_64)
GCC version: (Ubuntu 11.4.0-1ubuntu1~22.04) 11.4.0
Clang version: Could not collect
CMake version: version 3.22.1
Libc version: glibc-2.35

Python version: 3.10.0 (default, Mar  3 2022, 09:58:08) [GCC 7.5.0] (64-bit runtime)
Python platform: Linux-5.15.0-112-generic-x86_64-with-glibc2.35
Is CUDA available: True
CUDA runtime version: Could not collect
CUDA_MODULE_LOADING set to: LAZY
GPU models and configuration: 
GPU 0: NVIDIA H100 80GB HBM3
GPU 1: NVIDIA H100 80GB HBM3
GPU 2: NVIDIA H100 80GB HBM3
GPU 3: NVIDIA H100 80GB HBM3
GPU 4: NVIDIA H100 80GB HBM3
GPU 5: NVIDIA H100 80GB HBM3
GPU 6: NVIDIA H100 80GB HBM3
GPU 7: NVIDIA H100 80GB HBM3

Nvidia driver version: 550.90.07
cuDNN version: Could not collect
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: True

CPU:
Architecture:                       x86_64
CPU op-mode(s):                     32-bit, 64-bit
Address sizes:                      46 bits physical, 57 bits virtual
Byte Order:                         Little Endian
CPU(s):                             104
On-line CPU(s) list:                0-103
Vendor ID:                          GenuineIntel
Model name:                         Intel(R) Xeon(R) Platinum 8470
CPU family:                         6
Model:                              143
Thread(s) per core:                 1
Core(s) per socket:                 52
Socket(s):                          2
Stepping:                           8
BogoMIPS:                           4000.00
Flags:                              fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc art arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc cpuid aperfmperf tsc_known_freq pni pclmulqdq dtes64 monitor ds_cpl smx est tm2 ssse3 sdbg fma cx16 xtpr pdcm pcid dca sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand lahf_lm abm 3dnowprefetch cpuid_fault epb cat_l3 cat_l2 cdp_l3 invpcid_single cdp_l2 ssbd mba ibrs ibpb stibp ibrs_enhanced fsgsbase tsc_adjust bmi1 avx2 smep bmi2 erms invpcid cqm rdt_a avx512f avx512dq rdseed adx smap avx512ifma clflushopt clwb intel_pt avx512cd sha_ni avx512bw avx512vl xsaveopt xsavec xgetbv1 xsaves cqm_llc cqm_occup_llc cqm_mbm_total cqm_mbm_local split_lock_detect avx_vnni avx512_bf16 wbnoinvd dtherm ida arat pln pts avx512vbmi umip pku ospke waitpkg avx512_vbmi2 gfni vaes vpclmulqdq avx512_vnni avx512_bitalg tme avx512_vpopcntdq la57 rdpid bus_lock_detect cldemote movdiri movdir64b enqcmd fsrm md_clear serialize tsxldtrk pconfig arch_lbr amx_bf16 avx512_fp16 amx_tile amx_int8 flush_l1d arch_capabilities
L1d cache:                          4.9 MiB (104 instances)
L1i cache:                          3.3 MiB (104 instances)
L2 cache:                           208 MiB (104 instances)
L3 cache:                           210 MiB (2 instances)
NUMA node(s):                       8
NUMA node0 CPU(s):                  0-12
NUMA node1 CPU(s):                  13-25
NUMA node2 CPU(s):                  26-38
NUMA node3 CPU(s):                  39-51
NUMA node4 CPU(s):                  52-64
NUMA node5 CPU(s):                  65-77
NUMA node6 CPU(s):                  78-90
NUMA node7 CPU(s):                  91-103
Vulnerability Gather data sampling: Not affected
Vulnerability Itlb multihit:        Not affected
Vulnerability L1tf:                 Not affected
Vulnerability Mds:                  Not affected
Vulnerability Meltdown:             Not affected
Vulnerability Mmio stale data:      Not affected
Vulnerability Retbleed:             Not affected
Vulnerability Spec rstack overflow: Not affected
Vulnerability Spec store bypass:    Mitigation; Speculative Store Bypass disabled via prctl and seccomp
Vulnerability Spectre v1:           Mitigation; usercopy/swapgs barriers and __user pointer sanitization
Vulnerability Spectre v2:           Mitigation; Enhanced IBRS; IBPB conditional; RSB filling; PBRSB-eIBRS SW sequence; BHI BHI_DIS_S
Vulnerability Srbds:                Not affected
Vulnerability Tsx async abort:      Not affected

Versions of relevant libraries:
[pip3] numpy==1.26.4
[pip3] nvidia-cublas-cu12==12.4.5.8
[pip3] nvidia-cuda-cupti-cu12==12.4.127
[pip3] nvidia-cuda-nvrtc-cu12==12.4.127
[pip3] nvidia-cuda-runtime-cu12==12.4.127
[pip3] nvidia-cudnn-cu12==9.1.0.70
[pip3] nvidia-cufft-cu12==11.2.1.3
[pip3] nvidia-curand-cu12==10.3.5.147
[pip3] nvidia-cusolver-cu12==11.6.1.9
[pip3] nvidia-cusparse-cu12==12.3.1.170
[pip3] nvidia-cusparselt-cu12==0.6.2
[pip3] nvidia-ml-py==12.570.86
[pip3] nvidia-nccl-cu12==2.21.5
[pip3] nvidia-nvjitlink-cu12==12.4.127
[pip3] nvidia-nvtx-cu12==12.4.127
[pip3] pynvml==12.0.0
[pip3] pyzmq==26.3.0
[pip3] torch==2.6.0
[pip3] torchaudio==2.6.0
[pip3] torchvision==0.21.0
[pip3] transformers==4.50.0.dev0
[pip3] triton==3.2.0
[conda] numpy                     1.26.4                   pypi_0    pypi
[conda] nvidia-cublas-cu12        12.4.5.8                 pypi_0    pypi
[conda] nvidia-cuda-cupti-cu12    12.4.127                 pypi_0    pypi
[conda] nvidia-cuda-nvrtc-cu12    12.4.127                 pypi_0    pypi
[conda] nvidia-cuda-runtime-cu12  12.4.127                 pypi_0    pypi
[conda] nvidia-cudnn-cu12         9.1.0.70                 pypi_0    pypi
[conda] nvidia-cufft-cu12         11.2.1.3                 pypi_0    pypi
[conda] nvidia-curand-cu12        10.3.5.147               pypi_0    pypi
[conda] nvidia-cusolver-cu12      11.6.1.9                 pypi_0    pypi
[conda] nvidia-cusparse-cu12      12.3.1.170               pypi_0    pypi
[conda] nvidia-cusparselt-cu12    0.6.2                    pypi_0    pypi
[conda] nvidia-ml-py              12.570.86                pypi_0    pypi
[conda] nvidia-nccl-cu12          2.21.5                   pypi_0    pypi
[conda] nvidia-nvjitlink-cu12     12.4.127                 pypi_0    pypi
[conda] nvidia-nvtx-cu12          12.4.127                 pypi_0    pypi
[conda] pynvml                    12.0.0                   pypi_0    pypi
[conda] pyzmq                     26.3.0                   pypi_0    pypi
[conda] torch                     2.6.0                    pypi_0    pypi
[conda] torchaudio                2.6.0                    pypi_0    pypi
[conda] torchvision               0.21.0                   pypi_0    pypi
[conda] transformers              4.50.0.dev0              pypi_0    pypi
[conda] triton                    3.2.0                    pypi_0    pypi
ROCM Version: Could not collect
Neuron SDK Version: N/A
vLLM Version: 0.7.4.dev477+g61c6a5a7
vLLM Build Flags:
CUDA Archs: Not Set; ROCm: Disabled; Neuron: Disabled
GPU Topology:
GPU0    GPU1    GPU2    GPU3    GPU4    GPU5    GPU6    GPU7    NIC0    NIC1    NIC2    NIC3    NIC4    NIC5    NIC6    NIC7    NIC8    NIC9     NIC10   NIC11   CPU Affinity    NUMA Affinity   GPU NUMA ID
GPU0     X      NV18    NV18    NV18    NV18    NV18    NV18    NV18    PIX     PIX     PIX     SYS     SYS     SYS     SYS     SYS     SYS      SYS     SYS     SYS     0-12    0               N/A
GPU1    NV18     X      NV18    NV18    NV18    NV18    NV18    NV18    SYS     SYS     SYS     PIX     SYS     SYS     SYS     SYS     SYS      SYS     SYS     SYS     26-38   2               N/A
GPU2    NV18    NV18     X      NV18    NV18    NV18    NV18    NV18    SYS     SYS     SYS     SYS     PIX     SYS     SYS     SYS     SYS      SYS     SYS     SYS     39-51   3               N/A
GPU3    NV18    NV18    NV18     X      NV18    NV18    NV18    NV18    SYS     SYS     SYS     SYS     SYS     PIX     SYS     SYS     SYS      SYS     SYS     SYS     13-25   1               N/A
GPU4    NV18    NV18    NV18    NV18     X      NV18    NV18    NV18    SYS     SYS     SYS     SYS     SYS     SYS     PIX     PIX     PIX      SYS     SYS     SYS     52-64   4               N/A
GPU5    NV18    NV18    NV18    NV18    NV18     X      NV18    NV18    SYS     SYS     SYS     SYS     SYS     SYS     SYS     SYS     SYS      PIX     SYS     SYS     78-90   6               N/A
GPU6    NV18    NV18    NV18    NV18    NV18    NV18     X      NV18    SYS     SYS     SYS     SYS     SYS     SYS     SYS     SYS     SYS      SYS     PIX     SYS     91-103  7               N/A
GPU7    NV18    NV18    NV18    NV18    NV18    NV18    NV18     X      SYS     SYS     SYS     SYS     SYS     SYS     SYS     SYS     SYS      SYS     SYS     PIX     65-77   5               N/A
NIC0    PIX     SYS     SYS     SYS     SYS     SYS     SYS     SYS      X      PIX     PIX     SYS     SYS     SYS     SYS     SYS     SYS      SYS     SYS     SYS
NIC1    PIX     SYS     SYS     SYS     SYS     SYS     SYS     SYS     PIX      X      PIX     SYS     SYS     SYS     SYS     SYS     SYS      SYS     SYS     SYS
NIC2    PIX     SYS     SYS     SYS     SYS     SYS     SYS     SYS     PIX     PIX      X      SYS     SYS     SYS     SYS     SYS     SYS      SYS     SYS     SYS
NIC3    SYS     PIX     SYS     SYS     SYS     SYS     SYS     SYS     SYS     SYS     SYS      X      SYS     SYS     SYS     SYS     SYS      SYS     SYS     SYS
NIC4    SYS     SYS     PIX     SYS     SYS     SYS     SYS     SYS     SYS     SYS     SYS     SYS      X      SYS     SYS     SYS     SYS      SYS     SYS     SYS
NIC5    SYS     SYS     SYS     PIX     SYS     SYS     SYS     SYS     SYS     SYS     SYS     SYS     SYS      X      SYS     SYS     SYS      SYS     SYS     SYS
NIC6    SYS     SYS     SYS     SYS     PIX     SYS     SYS     SYS     SYS     SYS     SYS     SYS     SYS     SYS      X      PIX     PIX      SYS     SYS     SYS
NIC7    SYS     SYS     SYS     SYS     PIX     SYS     SYS     SYS     SYS     SYS     SYS     SYS     SYS     SYS     PIX      X      PIX      SYS     SYS     SYS
NIC8    SYS     SYS     SYS     SYS     PIX     SYS     SYS     SYS     SYS     SYS     SYS     SYS     SYS     SYS     PIX     PIX      X       SYS     SYS     SYS
NIC9    SYS     SYS     SYS     SYS     SYS     PIX     SYS     SYS     SYS     SYS     SYS     SYS     SYS     SYS     SYS     SYS     SYS       X      SYS     SYS
NIC10   SYS     SYS     SYS     SYS     SYS     SYS     PIX     SYS     SYS     SYS     SYS     SYS     SYS     SYS     SYS     SYS     SYS      SYS      X      SYS
NIC11   SYS     SYS     SYS     SYS     SYS     SYS     SYS     PIX     SYS     SYS     SYS     SYS     SYS     SYS     SYS     SYS     SYS      SYS     SYS      X 

Legend:

  X    = Self
  SYS  = Connection traversing PCIe as well as the SMP interconnect between NUMA nodes (e.g., QPI/UPI)
  NODE = Connection traversing PCIe as well as the interconnect between PCIe Host Bridges within a NUMA node
  PHB  = Connection traversing PCIe as well as a PCIe Host Bridge (typically the CPU)
  PXB  = Connection traversing multiple PCIe bridges (without traversing the PCIe Host Bridge)
  PIX  = Connection traversing at most a single PCIe bridge
  NV#  = Connection traversing a bonded set of # NVLinks

NIC Legend:

  NIC0: mlx5_0
  NIC1: mlx5_1
  NIC2: mlx5_2
  NIC3: mlx5_3
  NIC4: mlx5_4
  NIC5: mlx5_5
  NIC6: mlx5_6
  NIC7: mlx5_7
  NIC8: mlx5_8
  NIC9: mlx5_9
  NIC10: mlx5_10
  NIC11: mlx5_11

NCCL_CUMEM_ENABLE=0
TORCHINDUCTOR_COMPILE_THREADS=1
CUDA_MODULE_LOADING=LAZY
```


</details>


### 🐛 Describe the bug

For gemma-3-72b, llm.chat() works with a single message, but fails when inputting multiple messages (where each message can have multiple images).

```
Environment steps 1
INFO 03-16 00:20:22 [chat_utils.py:346] Detected the chat template content format to be 'openai'. You can set `--chat-template-content-format` to override this.
[rank0]:W0316 00:20:26.470743 178163 site-packages/torch/_dynamo/convert_frame.py:906] [0/8] torch._dynamo hit config.cache_size_limit (8)
[rank0]:W0316 00:20:26.470743 178163 site-packages/torch/_dynamo/convert_frame.py:906] [0/8]    function: 'forward_static' (/home/agi/vllm/vllm/model_executor/layers/layernorm.py:170)
[rank0]:W0316 00:20:26.470743 178163 site-packages/torch/_dynamo/convert_frame.py:906] [0/8]    last reason: 0/0: tensor 'L['x']' size mismatch at index 0. expected 64, actual 1
[rank0]:W0316 00:20:26.470743 178163 site-packages/torch/_dynamo/convert_frame.py:906] [0/8] To log all recompilation reasons, use TORCH_LOGS="recompiles".
[rank0]:W0316 00:20:26.470743 178163 site-packages/torch/_dynamo/convert_frame.py:906] [0/8] To diagnose recompilation issues, see https://pytorch.org/docs/main/torch.compiler_troubleshooting.html.
                                                                                                                                        ERROR 03-16 00:20:26 [core.py:340] EngineCore hit an exception: Traceback (most recent call last):input: 0.00 toks/s, output: 0.00 toks/s]
ERROR 03-16 00:20:26 [core.py:340]   File "/home/agi/vllm/vllm/v1/engine/core.py", line 333, in run_engine_core
ERROR 03-16 00:20:26 [core.py:340]     engine_core.run_busy_loop()
ERROR 03-16 00:20:26 [core.py:340]   File "/home/agi/vllm/vllm/v1/engine/core.py", line 367, in run_busy_loop
ERROR 03-16 00:20:26 [core.py:340]     outputs = step_fn()
ERROR 03-16 00:20:26 [core.py:340]   File "/home/agi/vllm/vllm/v1/engine/core.py", line 192, in step
ERROR 03-16 00:20:26 [core.py:340]     output = self.model_executor.execute_model(scheduler_output)
ERROR 03-16 00:20:26 [core.py:340]   File "/home/agi/vllm/vllm/v1/executor/abstract.py", line 80, in execute_model
ERROR 03-16 00:20:26 [core.py:340]     output = self.collective_rpc("execute_model",
ERROR 03-16 00:20:26 [core.py:340]   File "/home/agi/vllm/vllm/executor/uniproc_executor.py", line 56, in collective_rpc
ERROR 03-16 00:20:26 [core.py:340]     answer = run_method(self.driver_worker, method, args, kwargs)
ERROR 03-16 00:20:26 [core.py:340]   File "/home/agi/vllm/vllm/utils.py", line 2216, in run_method
ERROR 03-16 00:20:26 [core.py:340]     return func(*args, **kwargs)
ERROR 03-16 00:20:26 [core.py:340]   File "/home/agi/miniconda3/envs/pae/lib/python3.10/site-packages/torch/utils/_contextlib.py", line 116, in decorate_context
ERROR 03-16 00:20:26 [core.py:340]     return func(*args, **kwargs)
ERROR 03-16 00:20:26 [core.py:340]   File "/home/agi/vllm/vllm/v1/worker/gpu_worker.py", line 242, in execute_model
ERROR 03-16 00:20:26 [core.py:340]     output = self.model_runner.execute_model(scheduler_output)
ERROR 03-16 00:20:26 [core.py:340]   File "/home/agi/miniconda3/envs/pae/lib/python3.10/site-packages/torch/utils/_contextlib.py", line 116, in decorate_context
ERROR 03-16 00:20:26 [core.py:340]     return func(*args, **kwargs)
ERROR 03-16 00:20:26 [core.py:340]   File "/home/agi/vllm/vllm/v1/worker/gpu_model_runner.py", line 990, in execute_model
ERROR 03-16 00:20:26 [core.py:340]     hidden_states = self.model(
ERROR 03-16 00:20:26 [core.py:340]   File "/home/agi/miniconda3/envs/pae/lib/python3.10/site-packages/torch/nn/modules/module.py", line 1739, in _wrapped_call_impl
ERROR 03-16 00:20:26 [core.py:340]     return self._call_impl(*args, **kwargs)
ERROR 03-16 00:20:26 [core.py:340]   File "/home/agi/miniconda3/envs/pae/lib/python3.10/site-packages/torch/nn/modules/module.py", line 1750, in _call_impl
ERROR 03-16 00:20:26 [core.py:340]     return forward_call(*args, **kwargs)
ERROR 03-16 00:20:26 [core.py:340]   File "/home/agi/vllm/vllm/model_executor/models/gemma3_mm.py", line 531, in forward
ERROR 03-16 00:20:26 [core.py:340]     hidden_states = self.language_model.model(input_ids,
ERROR 03-16 00:20:26 [core.py:340]   File "/home/agi/vllm/vllm/compilation/decorators.py", line 172, in __call__
ERROR 03-16 00:20:26 [core.py:340]     return self.forward(*args, **kwargs)
ERROR 03-16 00:20:26 [core.py:340]   File "/home/agi/vllm/vllm/model_executor/models/gemma3.py", line 389, in forward
ERROR 03-16 00:20:26 [core.py:340]     hidden_states, residual = layer(
ERROR 03-16 00:20:26 [core.py:340]   File "/home/agi/miniconda3/envs/pae/lib/python3.10/site-packages/torch/nn/modules/module.py", line 1739, in _wrapped_call_impl
ERROR 03-16 00:20:26 [core.py:340]     return self._call_impl(*args, **kwargs)
ERROR 03-16 00:20:26 [core.py:340]   File "/home/agi/miniconda3/envs/pae/lib/python3.10/site-packages/torch/nn/modules/module.py", line 1750, in _call_impl
ERROR 03-16 00:20:26 [core.py:340]     return forward_call(*args, **kwargs)
ERROR 03-16 00:20:26 [core.py:340]   File "/home/agi/vllm/vllm/model_executor/models/gemma3.py", line 319, in forward
ERROR 03-16 00:20:26 [core.py:340]     hidden_states = self.self_attn(
ERROR 03-16 00:20:26 [core.py:340]   File "/home/agi/miniconda3/envs/pae/lib/python3.10/site-packages/torch/nn/modules/module.py", line 1739, in _wrapped_call_impl
ERROR 03-16 00:20:26 [core.py:340]     return self._call_impl(*args, **kwargs)
ERROR 03-16 00:20:26 [core.py:340]   File "/home/agi/miniconda3/envs/pae/lib/python3.10/site-packages/torch/nn/modules/module.py", line 1750, in _call_impl
ERROR 03-16 00:20:26 [core.py:340]     return forward_call(*args, **kwargs)
ERROR 03-16 00:20:26 [core.py:340]   File "/home/agi/vllm/vllm/model_executor/models/gemma3.py", line 190, in forward
ERROR 03-16 00:20:26 [core.py:340]     attn_output = self.attn(q, k, v)
ERROR 03-16 00:20:26 [core.py:340]   File "/home/agi/miniconda3/envs/pae/lib/python3.10/site-packages/torch/nn/modules/module.py", line 1739, in _wrapped_call_impl
ERROR 03-16 00:20:26 [core.py:340]     return self._call_impl(*args, **kwargs)
ERROR 03-16 00:20:26 [core.py:340]   File "/home/agi/miniconda3/envs/pae/lib/python3.10/site-packages/torch/nn/modules/module.py", line 1750, in _call_impl
ERROR 03-16 00:20:26 [core.py:340]     return forward_call(*args, **kwargs)
ERROR 03-16 00:20:26 [core.py:340]   File "/home/agi/vllm/vllm/attention/layer.py", line 214, in forward
ERROR 03-16 00:20:26 [core.py:340]     torch.ops.vllm.unified_attention_with_output(
ERROR 03-16 00:20:26 [core.py:340]   File "/home/agi/miniconda3/envs/pae/lib/python3.10/site-packages/torch/_ops.py", line 1123, in __call__
ERROR 03-16 00:20:26 [core.py:340]     return self._op(*args, **(kwargs or {}))
ERROR 03-16 00:20:26 [core.py:340]   File "/home/agi/vllm/vllm/attention/layer.py", line 363, in unified_attention_with_output
ERROR 03-16 00:20:26 [core.py:340]     self.impl.forward(self,
ERROR 03-16 00:20:26 [core.py:340]   File "/home/agi/vllm/vllm/v1/attention/backends/flash_attn.py", line 286, in forward
ERROR 03-16 00:20:26 [core.py:340]     cascade_attention(
ERROR 03-16 00:20:26 [core.py:340]   File "/home/agi/vllm/vllm/v1/attention/backends/flash_attn.py", line 397, in cascade_attention
ERROR 03-16 00:20:26 [core.py:340]     assert sliding_window == (-1, -1), (
ERROR 03-16 00:20:26 [core.py:340] AssertionError: Cascade attention does not support sliding window.
```

@DarkLight1337 

### Before submitting a new issue...

- [x] Make sure you already searched for relevant issues, and asked the chatbot living at the bottom right corner of the [documentation page](https://docs.vllm.ai/en/latest/), which can answer lots of frequently asked questions.

I used an OpenAI-Compatible Server with vllm serve and the model google/gemma-3-27b-it, but I encountered the same exact error. The model successfully started serving, but when I ran the client code, the error appeared, and the server crashed with the following message:

ERROR 03-16 23:08:33 [core.py:340] AssertionError: Cascade attention does not support sliding window.

I also tested the server using cURL with the same prompt, and it worked without any issues. However, when running the client code, the server crashed.

My vLLM version is 0.7.3.

--- a/vllm/v1/worker/gpu_model_runner.py
+++ b/vllm/v1/worker/gpu_model_runner.py
@@ -82,8 +82,15 @@ class GPUModelRunner(LoRAModelRunnerMixin):
             self.kv_cache_dtype = STR_DTYPE_TO_TORCH_DTYPE[
                 cache_config.cache_dtype]
 
-        self.is_multimodal_model = model_config.is_multimodal_model
+        # NOTE(woosuk): sliding_window is None for models with interleaved
+        # attention. Use interleaved_sliding_window instead.
         self.sliding_window = model_config.get_sliding_window()
+        self.interleaved_sliding_window = getattr(
+            model_config.hf_text_config, "interleaved_sliding_window", None)
+        self.window_size = (self.sliding_window
+                            or self.interleaved_sliding_window)
+
+        self.is_multimodal_model = model_config.is_multimodal_model
         self.block_size = cache_config.block_size
         self.max_model_len = model_config.max_model_len
         self.max_num_blocks_per_req = cdiv(self.max_model_len, self.block_size)
@@ -674,7 +681,7 @@ class GPUModelRunner(LoRAModelRunnerMixin):
             num_query_heads=self.num_query_heads,
             num_kv_heads=self.num_kv_heads,
             use_alibi=False,  # FIXME
-            use_sliding_window=self.sliding_window is not None,
+            use_sliding_window=self.window_size is not None,
             num_sms=self.num_sms,
         )
         return common_prefix_len if use_cascade else 0

